### ISSUE TITLE
cudaCheckError() failed : no kernel image is available for execution on the device

### ISSUE THREAD

**Issue by zhangguanghui1** [Sekunde/3D-SIS#13]
When I run "python main.py --cfg ScanNet/example --step 1205541 --output_dir example_result/scannet_benchmark_checkpoint --mode benchmark --gpu 0" from " example.sh.".
Ｉ have the problem:
...
 loaded mode from example_result/scannet_benchmark_checkpoint/ScanNet/example/checkpoints/step_1205541.pth
starting test on whole scan....
  0%|                                                     | 0/5 [00:00<?, ?it/s]/media/zgh/winz/3D/3D-SIS/lib/nets/network.py:546: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  rpn_cls_prob_level1 = F.softmax(rpn_cls_score_level1)
/media/zgh/winz/3D/3D-SIS/lib/nets/network.py:559: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  rpn_cls_prob_level2 = F.softmax(rpn_cls_score_level2)
**cudaCheckError() failed : no kernel image is available for execution on the device**

I do not how to solve it? and I re-run the build.py in nms and roi_pooling, it does not work.
Can you help me ？　Thank you very much.

**Comment by bonbonjour** [comment]
I got the same problem as you, if you have solved the problem, please tell me your solution, thanks!

**Comment by zhangguanghui1** [comment]
@bonbonjour Just re-complie the nms and roi_pooling.

**Comment by bonbonjour** [comment]
Do you mean running "nvcc -c -o nms_kernel.cu.o nms_kernel.cu -x cu -Xcompiler -fPIC -arch=sm_61"under nms and roi_pooling?

**Comment by bonbonjour** [comment]
can you please tell me the exactly steps to re-compile the nms and roi_pooling?

**Comment by Sekunde** [comment]
@bonbonjour  after nvcc compile, you still need
python build.py
checkout this:
https://github.com/Sekunde/3D-SIS/blob/master/lib/layer_utils/roi_pooling/build.py

**Comment by bonbonjour** [comment]
@Sekunde following your step, after running"python build.py", I got this:
/home/3D-SIS/lib/layer_utils/nms
/home/miniconda3/lib/python3.6/distutils/extension.py:131: UserWarning: Unknown Extension options: 'define_marcos'
  warnings.warn(msg)
generating /tmp/tmpvioy9uvy/_nms.c
setting the current directory to '/tmp/tmpvioy9uvy'
running build_ext
building '_nms' extension
creating home
creating home
creating home/3D-SIS
creating home/3D-SIS/lib
creating home/3D-SIS/lib/layer_utils
creating home/3D-SIS/lib/layer_utils/nms
creating home/3D-SIS/lib/layer_utils/nms/src
gcc -pthread -B /home/miniconda3/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/miniconda3/lib/python3.6/site-packages/torch/utils/ffi/../../lib/include -I/home/miniconda3/lib/python3.6/site-packages/torch/utils/ffi/../../lib/include/TH -I/home/miniconda3/lib/python3.6/site-packages/torch/utils/ffi/../../lib/include/THC -I/usr/local/cuda/include -I/home/miniconda3/include/python3.6m -c _nms.c -o ./_nms.o -std=c99
gcc -pthread -B /home/miniconda3/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/miniconda3/lib/python3.6/site-packages/torch/utils/ffi/../../lib/include -I/home/miniconda3/lib/python3.6/site-packages/torch/utils/ffi/../../lib/include/TH -I/home/miniconda3/lib/python3.6/site-packages/torch/utils/ffi/../../lib/include/THC -I/usr/local/cuda/include -I/home/miniconda3/include/python3.6m -c /home/3D-SIS/lib/layer_utils/nms/src/nms_cuda.c -o ./home/3D-SIS/lib/layer_utils/nms/src/nms_cuda.o -std=c99
/home/3D-SIS/lib/layer_utils/nms/src/nms_cuda.c: In function ‘gpu_nms’:
/home/3D-SIS/lib/layer_utils/nms/src/nms_cuda.c:24:37: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
     unsigned long long* mask_flat = THCudaLongTensor_data(state, mask);
                                     ^
/home/3D-SIS/lib/layer_utils/nms/src/nms_cuda.c:35:42: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
     unsigned long long * mask_cpu_flat = THLongTensor_data(mask_cpu);
                                          ^
/home/3D-SIS/lib/layer_utils/nms/src/nms_cuda.c:38:41: warning: initialization from incompatible pointer type [-Wincompatible-pointer-types]
     unsigned long long* remv_cpu_flat = THLongTensor_data(remv_cpu);
                                         ^
/home/3D-SIS/lib/layer_utils/nms/src/nms_cuda.c:17:9: warning: unused variable ‘boxes_dim’ [-Wunused-variable]
     int boxes_dim = THCudaTensor_size(state, boxes, 1);
         ^
gcc -pthread -shared -B /home/miniconda3/compiler_compat -L/home/miniconda3/lib -Wl,-rpath=/home/miniconda3/lib -Wl,--no-as-needed -Wl,--sysroot=/ ./_nms.o ./home/3D-SIS/lib/layer_utils/nms/src/nms_cuda.o /home/3D-SIS/lib/layer_utils/nms/src/cuda/nms_kernel.cu.o -o ./_nms.so

I still got the same error"cudaCheckError() failed : no kernel image is available for execution on the device"

**Comment by zhangguanghui1** [comment]
@bonbonjour Change the "-arch=sm_61" to suit your system or delete it directly.

**Comment by santhu937** [comment]
@zhangguanghui1  can u please suggest "-arch=sm_61" to suit for google colab

Tue Feb 18 08:14:49 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.48.02    Driver Version: 418.67       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla P100-PCIE...  Off  | 00000000:00:04.0 Off |                    0 |
| N/A   35C    P0    26W / 250W |      0MiB / 16280MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
